### ISSUE TITLE
nodejs interface

### ISSUE THREAD

**Issue by sambP** [LedgerHQ/xpub-scan#57]
Is there any way to use the scan feature directly via nodejs?

**Comment by pglushchenko-ledger** [comment]
I would vote to have this feature as well

**Comment by meriadec** [comment]
I would say the current codebase should be refactored to make this smooth. Today you could probably hack thing around like

```js
const { init } = require("@ledgerhq/xpub-scan/lib/helpers");
const checkBalance = require("@ledgerhq/xpub-scan/lib/actions/checkBalance");

async function main() {
  const xpub =
    "tpubDCR8dij5wujL48XPKtQATpKUEQkJL2VkuKrtypXXba7hWSjW8uziMYVud6HHT99egY4BG1WUJ3z8hcDvdwrWGvTuk8AfbqF6euw1F2gRvUj";

  const silent = false;
  const quiet = false;
  const currency = "BTC";
  const testnet = undefined;
  const derivationMode = undefined;

  init(xpub, silent, quiet, currency, testnet, derivationMode);

  const res = await checkBalance.run(xpub);

  console.log(res);
}

main()
```

**Comment by glethuillier** [comment]
This is a great idea. It will require some refactoring. We will definitely take a look at this, @sambP.

**Comment by sambP** [comment]
Thanks @meriadec, works so far. The only problem is, that I have to scan multiple pubs in parallel, it would be nice to get an instance of init() somehow.  

Sounds great, @glethuillier. I know everyone hates time estimates, but would it be possible to give me one? :) We need this feature for our product and I have to decide if I just wait or try to build a solution for myself.

**Comment by sambP** [comment]
I checked the code, running checkBalance in parellel with different xpubs seems to work fine, as long as they can share the global configuration. We use only btc xpubs, so no problems for us and no time estimates anymore needed :)

Looking forward for an official solution.

**Comment by sambP** [comment]
@meriadec sorry but one question a little bit offtopic, do I get with the checkBalance.run function also addresses which are currently empty but hold assets in the past? I would need them to find all movements.

**Comment by glethuillier** [comment]
@sambP Yes, this function returns all addresses with at least one operation. Therefore, you will have all past movements.

**Comment by meriadec** [comment]
I think @glethuillier should be able to answer as he's the author of the lib, I would rather not saying sh*t :smile: 

To answer your previous questions about how the code will evolve in the future, I don't think we can give you satisfying answers for now. This tool has been initially built for internal QA purposes first, although we already considered splitting in into more manageable & documented pieces (e.g: balance module, comparison module..) but it will probably take time.

But feedbacks always appreciated :)
